src/components/FilmCard.tsx

```diff
@@ -0,0 +1,120 @@
+import { useContext, useEffect, useState } from 'react';
+import { Rate, Tag } from 'antd';
+import { HeartFilled, StarFilled} from '@ant-design/icons';
+import { LangContext } from '../utils/LangContext';
+import { getTranslation } from '../utils/getTranslation';
+import { Film } from '../types/Film';
+import { genreColor } from '../utils/genreColor';
+import '../App.scss';
+import { Genre } from '../types/Genre';
+import { fetchData } from '../utils/fetchData';
+import { genreURL, posterURL } from '../utils/filmUrlSettings';
+
+type Props = {
+  film: Film | null,
+  className: string,
+}
+
+export const FilmCard: React.FC<Props> = ({ film, className }) => {
+  const lang = useContext(LangContext);
+  const [genres, setGenres] = useState<Genre[]>([]);
+  
+  async function getGenres(lang: string) {
+    try {
+      const data = await fetchData(genreURL + '&language=' + lang, 'genres');
+      setGenres(data);
+    } catch {
+      setGenres([]);
+    }
+  }
+  
+  useEffect(() => {
+    getGenres(lang);
+  }, [lang]);
+
+  return (
+    <>
+      {film && ( 
```

Comment:
you can use early exit approach
```suggestion
  if (!film) {
    return (
      <p>Film not found</p>
    );
  }

  return (
```